### ISSUE TITLE
Feature Request: GGML_OP_UNIFORM_RANDOM, or another way to generate random numbers on-device

### ISSUE THREAD

**Issue by cdanis** [ggml-org/llama.cpp#28974]
### Prerequisites

- [x] I am running the latest code. Mention the version if possible as well.
- [x] I carefully followed the [README.md](https://github.com/ggml-org/llama.cpp/blob/master/README.md).
- [x] I searched using keywords relevant to my issue to make sure that I am creating a new issue that is not already open (or closed).
- [x] I reviewed the [Discussions](https://github.com/ggml-org/llama.cpp/discussions), and have a new and useful enhancement to share.

### Feature Description

Allow for on-GPU and in-graph computation of random numbers.

I think this likely makes the most sense as a new ggml op like GGML_OP_UNIFORM_RANDOM (initialized with a seed param), but I can imagine backend tensor fill functions making sense as well.

### Motivation

I'm working on a llama.cpp port of Gumbel-Softmax Quantization ([arXiv 2604.18556](https://arxiv.org/abs/2604.18556) & [IST-DASLab/GSQ](https://github.com/IST-DASLab/GSQ)).  It's a cool technique for squeezing more accuracy out of quantizations, and can be directly applied to many of the existing quant schemes, with large gains on low-bit formats.

I need a fresh tensor of uniform [0,1) i.i.d. floats on the GPU at every optimizer step (hundreds to thousands).  The values must be a deterministic function of (seed, flat index) for reproducibility.  There's no GGML OP for randomness, so in my research fork I implemented a CUDA kernel plus CPU fallback.  (I tried plain CPU rng,  but the optimizer time was dominated by the generation + H2D copy.)

A generic solution wouldn't just be useful for GSQ, I think it has broader applicability: examples/diffusion already uses Gumbel noise, presently generated CPU-side only.  Other techniques like QAT (w/ stochastic rounding) or differentiable k-means / top-k could use the same new randomness op, as could VAE training or diffusion model training.

### Possible Implementation

CPU implementation can just use mt19937 as usual.

On-device we could use something like Xoroshiro128+ or Philox.

I'm happy to attempt a full implementation.
